warehouse/ingest-scripts/src/main/resources/bin/util/buildIndexStats.sh

```diff
@@ -0,0 +1,365 @@
+#
+# Precision values should be set based upon memory usage versus error percentage. Several
+# tests indicate that inconsistent results can occur when the sparse and normal precision
+# values differ. The greater the difference, the larger the error percentage. While most
+# of the generated cardinalities fell within the projected error percentage, in the few
+# instances where it did it was off by very significat factor. Thus when selecting
+# sparse and normal precision values it is best to chose the same value for both.
+#
+#+++++++++++++++++++++++++++++++++++++++
+# Optional Debug Environment Variables
+# There are several options that will allow for analysis of the debug data:
+#   _UniqueCounts => Collects actual unique counts in addition to the HyperLogLogPlus
+#           generated cardinality for analysis purposes. The mapper will generate the
+#           the unique count, while the reducer is the best location to view all of the
+#           entries from multiple mappers. The test script hlParse.sh is provided to
+#           parse the lo file of the reducer. This process extracts the analysis data
+#           into a csv format, which can be loaded into a spreadsheet for analyis.
+#   _GCMapperLog => Enables GC logging for the mapper.
+#   _GCReducerLog => Enables GC logging for the reducer.
+#   _GCLog => Enables GC logging for both the mapper and reducer.
+#   _GCLogDir => Specifies the directory for the GC logs (overrides the deault value).
+#   _HeapDump => Enable heap dump for mapper and reducer.
+#========================================================================================
+
+if [[ `uname` == "Darwin" ]]; then
+        THIS_SCRIPT=`python -c 'import os,sys;print os.path.realpath(sys.argv[1])' $0`
+else
+        THIS_SCRIPT=`readlink -f $0`
+fi
+
+THIS_DIR="${THIS_SCRIPT%/*}"
+cd $THIS_DIR
+
+. ../ingest/ingest-env.sh
+
+declare -r _base=$(basename $0)
+declare -r _lockFile=${LOCK_FILE_DIR}/${_base}
+
+
+#===================================================================================
+# local functions
+function cleanup() {
+    declare -r _status=$1
+
+    rm -f "${_lockFile}"
+    trap 0 1 2 3 15
+    exit ${_status}
+}
+
+function setupLockFile() {
+    # check for current execution of this script
+    test -e ${_lockFile} && {
+        echo "ERROR: lock file exists => ${_lockFile}"
+        exit 1
+    }
+    touch "${_lockFile}"
+    trap "cleanup 1" 1 2 3 15
+    trap "cleanup 0" 0
+}
+
+#==================================
+# Sets JVM parameters for Mapper
+function setMapperJVMParameters() {
+    local _MapperMemoryMB
+    if [[ -n "${_UniqueCounts}" ]]; then
+        # memory setting may need to be adjusted based upon input data
+        # check max container size
+        _MapperMemoryMB=1792
+    else
+        _MapperMemoryMB=1024
+    fi
+
+    # allocate 1/3 memory to mapper io sort buffer but muat be less than 2048
+    ((_MapInMemSortBufferMBSize = _MapperMemoryMB / 3))
```

Comment:
Guessing that `_MapInMemSortBufferMBSize` was intended to be `_MapInMemorySortBufferMBSize` instead(?) 

`_MapInMemorySortBufferMBSize` declared below but never used

Reply:
I had fixed this before. Somehow it got back in. fixed again